src/plugins/matrixops/constant.cpp

```diff
@@ -39,11 +39,11 @@ namespace phylanx { namespace execution_tree { namespace primitives
             ir::range const& shape)
         {
             std::array<std::size_t, 2> result = {0, 0};
-            result[0] = extract_integer_value(*shape.begin());
+            result[0] = extract_integer_value(*shape.begin())[0];
```

Comment:
We should probably throw an exception if `extract_integer_value` returns a `node_data<int64_t>` that holds anything but a scalar value (matrices and vectors don't make any sense). This probably applies for all uses of this function where the result is 'converted' to a single value.

Reply:
Better yet, we should introduce a new function `extract_scalar_integer_value` that does the proper handling and use that everywhere we currently use `[0]` on the result.